### ISSUE TITLE
[Minor Bug] Dropped items can be invisible even if still obtainable.

### ISSUE THREAD

**Issue by SatDog92** [sp614x/optifine#5349]
## Description of Issue
Dropped items become invisible (even if still obtainable). Can happen when the player throws one with Q or when they mine a block. The item is not displayed but it's still there, similar to invisible mobs. To my experience, this doesn't affect mob entity drops, only block entity ones.

## Steps to Reproduce
Mine blocks until one of them drop nothing (tested with dirt, happens with wood, ores and stone as well).

## OptiFine Version
OptiFine_1.16.5_HD_U_G7_pre11

## Installation Method
Both methods (as i used both Vanilla, Forge and Fabric)

## Fabric/Forge Version
Forge 36.0.42
Fabric 0.11.1, Fabric API 0.30.3, and Optifabric 1.9.2

## Other Installed Mods
None (besides Fabric API and Optifabric for Fabric Loader).

## Log Files/Crash Reports
Vanilla:
[vanilla log](https://github.com/sp614x/optifine/files/6022701/vanilla.log.log)
Forge:
[optifine-log.log](https://github.com/sp614x/optifine/files/6022617/optifine-log.log)
[optifine-debug.log](https://github.com/sp614x/optifine/files/6022619/optifine-debug.log)
Fabric:
[Optifine Fabric log](https://github.com/sp614x/optifine/files/6022556/Optifine.Fabric.log.log)

## F3 Debug Screenshot
Vanilla:
[Vanilla version 1](https://user-images.githubusercontent.com/56725716/108723946-174b9080-7525-11eb-94cd-f8380be13eeb.png)
[Vanilla version 2](https://user-images.githubusercontent.com/56725716/108723952-17e42700-7525-11eb-8b5b-3a6aecfd8050.png)
(you can see clearly that even if there was no dirt block item visible after mining a dirt block, i was still able to get it. Same happens on Forge and Fabric)
Forge:
[Forge version 1](https://user-images.githubusercontent.com/56725716/108721459-4f050900-7522-11eb-8a18-069770f15222.png)
[Forge version 2](https://user-images.githubusercontent.com/56725716/108721463-50363600-7522-11eb-9711-28950eb17638.png)
Fabric:
[Fabric version](https://user-images.githubusercontent.com/56725716/108721055-d900a200-7521-11eb-9285-d30ac6e056a8.png)

## Prior Testing
If you've installed using Forge/Fabric:
- Does this happen without other mods installed? 
   - Only OptiFine and Forge or only OptiFine, OptiFabric, and Fabric
- Does this happen without any mods at all?
   - Yes, only Forge/Fabric by itself.
- Does this happen in OptiFine standalone?
   - Yes.
- Does this happen in Vanilla without OptiFine?
   - No.

**Comment by Cart3r1234** [comment]
Can confirm this does happen on the latest releases of optifine 1.16.5

**Comment by Cart3r1234** [comment]
Still happens in the latest preview of G8

**Comment by sp614x** [comment]
Does it happen without shaders?

**Comment by SatDog92** [comment]
> Does it happen without shaders?

Yes. Maybe i didn't express it correctly after reading my OP:
> Happens with Optifine itself, no mods installed (aside from Optifabric for Fabric ) and active shaders.
I meant "no mods and no shaders". Sorry.

**Comment by sp614x** [comment]
Can't reproduce it, mined and later dropped several stacks of items.
Does it happen only when using enchanted tools?

**Comment by Cart3r1234** [comment]
I am able to reproduce it just by dropping an item, picking it up, and dropping it again.  After a few times one of them will be invisible, but still obtainable.

**Comment by SatDog92** [comment]
> Can't reproduce it, mined and later dropped several stacks of items.
> Does it happen only when using enchanted tools?

No. Indeed, when it first happened, i was in the first minutes of a new survival game, using a stone shovel. So, it happens in both enchanted and unenchanted tools.

Might it be possible it depends from the specs? While my PC is not NASA computer and has a pretty standard setup, it's still pretty good:
-Intel Core i9 9900K 3.60 GHz, and 32 RAM;
-Nvidia GeForce RTX 2060 6GB.

>I am able to reproduce it just by dropping an item, picking it up, and dropping it again. After a few times one of them will be invisible, but still obtainable.

This happens as well.

**Comment by sp614x** [comment]
Is the item appearing when looking around (moving the mouse)?

**Comment by Cart3r1234** [comment]
From my experience with this bug once the item goes invisible the only ways to make it reappear are to collect it, drop another item of that type near it so they stack together, or reload the world

**Comment by SatDog92** [comment]
> I managed to reproduce this for dropped items, but the item appears if I move the mouse minimally (look around).
> Is the invisible item appearing when looking around (moving the mouse)?

No. It's as if the item fails to render in that instance. Moreover, when you get the invisible item, it doesn't plop like normal, but it's completely silent.

After a few more testings, i'm pretty sure the item is rendered again by also going away and reloading the chunk the item is on, and not only by quitting the game. But for me, it doesn't reappear by looking around.

**Comment by SteamDoge9000** [comment]
Good thinkg I found this Im usinhg shaders and it seems to be happening

like its random, the only thing I noticed is that it seems to happen maybe when your close to a block wall or something, idk

**Comment by SteamDoge9000** [comment]
looking into the log and running a quick test, when the item does go invisable i think its producing this error

![image](https://user-images.githubusercontent.com/38770780/139516070-b3385902-22b9-4525-a145-593c0b62cdba.png)

**Comment by SteamDoge9000** [comment]
can confirm, after running another quick testr, the item went invisable again and it kicked up this message

so when the item drop dosent render, this error will kick up

![image](https://user-images.githubusercontent.com/38770780/139516125-9d240c8c-0a6d-4382-944d-927f83472659.png)

**Comment by SteamDoge9000** [comment]
it may seem like theres only 1 item, but I droped 2, one of turned invisable and while its still there and can be picked up, a "box" does not show up for it

![image](https://user-images.githubusercontent.com/38770780/139516294-efc589b5-5ecd-4df8-94d9-ce26f52754fb.png)
